### DISCUSSION TITLE
Aspectize via @ considerations

### DISCUSSION THREAD

**Discussion by amogorkon** [amogorkon/justuse#305] [Ideas]
so, i was thinking about aspectizing and @ operator for modules
i very much like the idea of adding `__matmul__` to ProxyModules, but i think if we do that and people realize you *can* do that with modules, they'll want to start using it too, which then leads to us shadowing them
so i thought maybe we could apply patma of sorts to the arguments passed into `__matmul__` and check if they match ours and otherwise pass along
to avoid collisions in the future inspired by us ^^

**Comment by amogorkon** [comment]
could be simplified to `mod @ (check, pattern, func)`

**Comment by greyblue9** [comment]
> so, i was thinking about aspectizing and @ operator for modules i very much like the idea of adding `__matmul__` to ProxyModules, but i think if we do that and people realize you _can_ do that with modules, they'll want to start using it too, which then leads to us shadowing them so i thought maybe we could apply patma of sorts to the arguments passed into `__matmul__` and check if they match ours and otherwise pass along to avoid collisions in the future inspired by us ^^

I'm not sure I understand the concern around shadowing?
